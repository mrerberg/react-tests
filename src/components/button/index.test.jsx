import { Button } from "./index";
import { screen, render, fireEvent } from "@testing-library/react";

const dataTestId = "test-id";

describe("<Button />", () => {
  describe("Snapshots tests", () => {
    it("should render successfully and match snapshot", () => {
      const { container } = render(<Button>Click me</Button>);

      expect(container).toMatchSnapshot();
    });
  });

  describe("Attributes tests", () => {
    it('should set type="button" by default', () => {
      render(<Button />);

      expect(screen.getByRole("button")).toHaveAttribute("type", "button");
    });

     it("should set type attribute", () => {
       const type = "submit";
       render(<Button type={type} />);

       expect(screen.getByRole("button")).toHaveAttribute("type", type);
     });
  });

  describe("Callbacks tests", () => {
     it("should call `onClick` prop", () => {
       const cb = jest.fn();

       render(
         <Button onClick={cb} dataTestId={dataTestId} />
       );

       fireEvent.click(screen.getByTestId(dataTestId));

       expect(cb).toBeCalledTimes(1);
     });
  });
});
