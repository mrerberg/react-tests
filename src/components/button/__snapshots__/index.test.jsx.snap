// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Snapshots tests should render successfully and match snapshot 1`] = `
<div>
  <button
    class="button"
    type="button"
  >
    Click me
  </button>
</div>
`;
